bin/Dockerfile

```diff
@@ -1,8 +1,18 @@
 FROM continuumio/miniconda3
 RUN conda install -c conda-forge fenics
-COPY . /tmp/mud-examples
+#COPY . /tmp/mud-examples
+RUN pip install -U setuptools setuptools_scm wheel
+COPY .git /tmp/mud-examples
+COPY src/ /tmp/mud-examples/src
+COPY README.md /tmp/mud-examples
+COPY MANIFEST.in /tmp/mud-examples
+COPY setup.py /tmp/mud-examples
+COPY setup.cfg /tmp/mud-examples
```

Comment:
this is preferable over line 3 because if there's a change to something like a Makefile, the build process can still leverage cached layers.